MediaBrowser.Model/Dlna/StreamBuilder.cs

```diff
@@ -771,10 +771,17 @@ private TranscodingProfile GetVideoTranscodeProfile(MediaSourceInfo item, VideoO
 
         private void BuildStreamVideoItem(StreamInfo playlistItem, VideoOptions options, MediaSourceInfo item, MediaStream videoStream, MediaStream audioStream, IEnumerable<MediaStream> candidateAudioStreams, string container, string videoCodec, string audioCodec)
         {
-            // prefer matching video codecs
             var videoCodecs = ContainerProfile.SplitValue(videoCodec);
             var directVideoCodec = ContainerProfile.ContainsContainer(videoCodecs, videoStream?.Codec) ? videoStream?.Codec : null;
-            playlistItem.VideoCodecs = directVideoCodec != null ? new[] { directVideoCodec } : videoCodecs;
+            if (directVideoCodec != null)
+            {
+                // merge directVideoCodec to videoCodecs
+                videoCodecs = videoCodecs != null && videoCodecs.Length > 0
+                    ? videoCodecs.Union(new[] { directVideoCodec }).ToArray()
+                    : new[] { directVideoCodec };
```

Comment:
```suggestion
                Array.Resize(ref videoCodecs, (videoCodecs?.Length ?? 0) + 1);
                videoCodecs[^1] = directVideoCodec;
```

I think this is more efficient?